valhalla/sif/dynamiccost.h

```diff
@@ -1021,7 +1026,8 @@ class DynamicCost {
     c.cost += destination_only_penalty_ * (edge->destonly() && !pred->destonly());
     c.cost +=
         alley_penalty_ * (edge->use() == baldr::Use::kAlley && pred->use() != baldr::Use::kAlley);
-    c.cost += maneuver_penalty_ * (!edge->link() && !edge->name_consistency(idx));
+    c.cost += maneuver_penalty_ * (!edge->link() && !edge->name_consistency(idx)) *
+              (is_hov_transition ? 0.1 : 1.0);
```

Comment:
its hard for me to imagine the scenario where this occurs. do you have an example?

Reply:
@kevinkreiser An HOV lane - in nearly every case I've studied - is just a lane change (a gentle turn; no need to slow down).  However, HOV lanes are also: 1) connected by a link, and 2) are named differently.

Here's a way for the link that connects "US 36" (not hov) to "US 36 Express" (hov):  https://www.openstreetmap.org/way/632700110

If I understand what I'm seeing in the debugger, Valhalla sees two non-trivial transitions (for what is otherwise really just a lane change):
1) from the highway onto the link
2) from the link onto the hov-lane

I've pushed some code changes that represent my latest thoughts (and appear to work).  However, I'm not sure if 1) I absolutely need to steal another bit from `EdgeLabel` to solve problem #2 above, and 2) I'm really not sure whether I've implemented `AutoCost::TransitionCostReverse` correctly (still getting my mind around the reverse stuff).

So yea, like I said, to me, transitioning onto/off-of an hov lane is just a simple lane change.  I'm just trying to figure out how to code it up.